runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -365,8 +365,33 @@ public Response getErroredStatesByTime(@PathParam("stateMachineName") String sta
         if(fromTimestamp.after(toTimestamp)) {
             return Response.status(Response.Status.BAD_REQUEST).entity("fromTime: " + fromTime + " should be before toTime: " + toTime).build();
         }
+        List<Status> statuses = Arrays.asList(Status.errored, Status.cancelled, Status.sidelined);
+        return Response.status(200).entity(statesDAO.findStatesByStatus(stateMachineName, fromTimestamp, toTimestamp, stateName, statuses)).build();
+    }
+
+    /**
+     * Retrieves all initialized states for the given range of time for a particular state machine name.
+     * @return json containing list of [state machine id, state id, status]
+     */
+    @GET
+    @Path("/{stateMachineName}/states/initializedbytime")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getInitializedStatesByTime(@PathParam("stateMachineName") String stateMachineName,
```

Comment:
Should I add an extra generic API, or replace these ones? I did consider the latter, but then it would break for people who are using the current errored API.